consensus/src/commons.rs

```diff
@@ -139,35 +141,29 @@ impl QuorumMsgSender {
     }
 
     /// Sends an quorum (internally) to the quorum loop.
-    pub(crate) async fn send(&self, msg: Message) -> bool {
-        if let Payload::Quorum(q) = &msg.payload {
-            if q.validation.is_empty()
-                || q.ratification.is_empty()
-                || q.vote == Vote::NoCandidate
+    pub(crate) async fn send(&self, msg: Message) {
+        match &msg.payload {
             // TODO: Change me accordingly to https://github.com/dusk-network/rusk/issues/1268
+            Payload::Quorum(q)
+                if !q.validation.is_empty()
+                    && !q.ratification.is_empty()
+                    && q.vote != Vote::NoCandidate =>
             {
-                return false;
+                tracing::debug!(
+                    event = "send quorum_msg",
+                    vote = %q.vote,
+                    round = msg.header.round,
+                    iteration = msg.header.iteration,
+                    validation = ?q.validation,
+                    ratification = ?q.ratification,
+                );
             }
-
-            tracing::debug!(
-                event = "send quorum_msg",
-                vote = %q.vote,
-                round = msg.header.round,
-                iteration = msg.header.iteration,
-                validation = ?q.validation,
-                ratification = ?q.ratification,
-            );
-
-            let _ = self
-                .queue
-                .send(msg.clone())
-                .await
-                .map_err(|e| error!("send quorum_msg failed with {:?}", e));
-
-            return true;
+            _ => return,
         }
 
-        false
+        if let Err(e) = self.queue.send(msg).await {
```

Comment:
I'd rather consider `oneshot::channel` instead of the bidirectional queue here (queue: AsyncQueue).

Reply:
Probably yes, as @ureeves mentioned, we should probably review all our queues.
For now, I've only refactored this code to not clone the msg without any architectural change